akka-http2-support/src/main/scala/akka/http/scaladsl/Http2.scala

```diff
@@ -60,36 +60,41 @@ class Http2Ext(private val config: Config)(implicit val system: ActorSystem) ext
 
     val effectivePort = if (port >= 0) port else 443
 
-    val serverLayer: BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed] = {
-      val wrappedContext = WrappedSslContextSPI.wrapContext(httpsContext)
-      val tls = http.sslTlsStage(wrappedContext, TLSRole.server)
+    def http2Layer(): BidiFlow[HttpResponse, SslTlsOutbound, SslTlsInbound, HttpRequest, NotUsed] =
+      Http2Blueprint.serverStack() atop
+        unwrapTls()
 
-      val unwrapTls: BidiFlow[ByteString, SslTlsOutbound, SslTlsInbound, ByteString, NotUsed] =
-        BidiFlow.fromFlows(Flow[ByteString].map(SendBytes), Flow[SslTlsInbound].collect {
-          case SessionBytes(_, bytes) ⇒ bytes
-        })
+    def unwrapTls(): BidiFlow[ByteString, SslTlsOutbound, SslTlsInbound, ByteString, NotUsed] =
+      BidiFlow.fromFlows(Flow[ByteString].map(SendBytes), Flow[SslTlsInbound].collect {
+        case SessionBytes(_, bytes) ⇒ bytes
+      })
 
-      Http2Blueprint.serverStack() atop
-        unwrapTls atop
+    @volatile var chosenProtocol: String = "h1"
```

Comment:
This is a local variable, no need for private. In fact, it doesn't even need to be volatile because there should already be a happens-before relationship enforced by sending out the first `SessionBytes`. I updated the code with assertions to make sure that it is not written more than once. Also added the comment that the flow cannot be reused.